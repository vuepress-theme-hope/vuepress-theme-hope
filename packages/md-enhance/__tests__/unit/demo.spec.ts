import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import {
  normalDemo,
  reactDemo,
  vueDemo,
} from "../../src/node/markdown-it/index.js";

describe("demo", () => {
  const markdownIt = MarkdownIt({ linkify: true })
    .use(normalDemo)
    .use(vueDemo)
    .use(reactDemo);

  it("Should resolve demo info", () => {
    expect(
      markdownIt.render(
        `
::: normal-demo A normal demo

\`\`\`html
<h1>Mr.Hope</h1>
<p>Is <span id="very">very</span> handsome</p>
\`\`\`

\`\`\`js
document.querySelector("#very").addEventListener("click", () => {
  alert("Very handsome!");
});
\`\`\`

\`\`\`css
span {
  color: red;
}
\`\`\`

:::

::: react-demo A react demo

\`\`\`js
export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { message: "very handsome" };
  }
  render() {
    return (
      <div className="box-react">
        Mr.Hope is <span>{this.state.message}</span>
      </div>
    );
  }
}
\`\`\`

\`\`\`css
.box-react span {
  color: red;
}
\`\`\`

:::

::: vue-demo A vue demo

\`\`\`vue
<template>
  <div class="box-vue">
    Mr.Hope is <span>{{ message }}</span>
  </div>
</template>
<script>
export default {
  data: () => ({ message: "very handsome" }),
};
</script>
<style>
.box-vue span {
  color: red;
}
</style>
\`\`\`

:::

::: normal-demo A normal demo

\`\`\`md
# Title

is very handsome.
\`\`\`

\`\`\`ts
const message: string = "Mr.Hope";

document.querySelector("h1").innerHTML = message;
\`\`\`

\`\`\`scss
h1 {
  font-style: italic;

  + p {
    color: red;
  }
}
\`\`\`

:::
`
      )
    ).toMatchSnapshot();
  });
});
