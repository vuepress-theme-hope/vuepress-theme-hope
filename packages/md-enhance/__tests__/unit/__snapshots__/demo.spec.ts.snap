// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`demo > Should resolve demo info 1`] = `
"
<CodeDemo id=\\"code-demo-0\\" type=\\"normal\\" title=\\"A%20normal%20demo\\" code=\\"eJw9jkELwjAMhf9KjBcFmXidXW+Cgp4ET72MNrBp1862G8jYfzed4CUJyfvey4RN6iyWKJqDvIXi7HsSe56VE728RBCxrx20plI4UvgolLmJfV5LaGpnou8Y6ZnAHT4jexmvh45cKt4Da+9kSScfNgrXP4ttURtzGllxbWMiR/mmbatfCnew2UIlYVIOoLYUEt8ejP2zVmxwVG7OhRN1zJHLlwujvfWhhEAmi1gyfwHzbUvd\\">
<pre><code class=\\"language-html\\">&lt;h1&gt;Mr.Hope&lt;/h1&gt;
&lt;p&gt;Is &lt;span id=&quot;very&quot;&gt;very&lt;/span&gt; handsome&lt;/p&gt;
</code></pre>
<pre><code class=\\"language-js\\">document.querySelector(&quot;#very&quot;).addEventListener(&quot;click&quot;, () =&gt; {
  alert(&quot;Very handsome!&quot;);
});
</code></pre>
<pre><code class=\\"language-css\\">span {
  color: red;
}
</code></pre>
</CodeDemo>
<CodeDemo id=\\"code-demo-5\\" type=\\"react\\" title=\\"A%20react%20demo\\" code=\\"eJxFkEFrwzAMhf/Kw6cOtvSepYWxyy7bYedcvERbMxLbSEpJCfnvtesmBWOQrPfeZ83mX0xpaAqeFS392rFXNL0VwVsIoEnJtYJvso0W734I3pFTzLUDGu9EeWzU8y6wD/KU+4CMgdbea27pqZNC1CrhgBkDidg/KlGbM/EFJxtj/EC1wXJTLOniGB6NNl8mHdlhlyugartzpv2yAx1q8+OnF06stTmuQ8AnFx8+EDpBJcG64/ygKe4kS7W/PW3W++h9r/IfIlE85tk0knZWbFlIwnUlvecycrZRkqaXKxQ1dTY=\\">
<pre><code class=\\"language-js\\">export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { message: &quot;very handsome&quot; };
  }
  render() {
    return (
      &lt;div className=&quot;box-react&quot;&gt;
        Mr.Hope is &lt;span&gt;{this.state.message}&lt;/span&gt;
      &lt;/div&gt;
    );
  }
}
</code></pre>
<pre><code class=\\"language-css\\">.box-react span {
  color: red;
}
</code></pre>
</CodeDemo>
<CodeDemo id=\\"code-demo-9\\" type=\\"vue\\" title=\\"A%20vue%20demo\\" code=\\"eJxNjsEOgjAQRH9l0pMmKnesnL34B71UuipJoU23EAjpv1uQEG+T3Zk3M4uhJ1EKGan1VkeqVAdI0wyorWa+KfF04zmblFhfwCNc7s4TGoZkr7tqntESs34TUpLFelspRcZkJYs/uOQ6ND5mRaN3IcLQS/c2Yl4iRkdd4nDErcJhx5ZQYqAw4aM7w67NW5COJ9Wl60LfiZLjZJeSy7YZy5YfuXbWhRKBTM6kNbaZRfoChEVaCQ==\\">
<pre><code class=\\"language-vue\\">&lt;template&gt;
  &lt;div class=&quot;box-vue&quot;&gt;
    Mr.Hope is &lt;span&gt;{{ message }}&lt;/span&gt;
  &lt;/div&gt;
&lt;/template&gt;
&lt;script&gt;
export default {
  data: () =&gt; ({ message: &quot;very handsome&quot; }),
};
&lt;/script&gt;
&lt;style&gt;
.box-vue span {
  color: red;
}
&lt;/style&gt;
</code></pre>
</CodeDemo>
<CodeDemo id=\\"code-demo-12\\" type=\\"normal\\" title=\\"A%20normal%20demo\\" code=\\"eJwtjb0KwkAQhF9lWBtFDdhG7C200nKbcFnNwWVPb1dBxHf3/IHpZub7njT21NIEx+hJWFmj4S7lgaHT3vIoDSstyK2uQlZzjGLWnaWFeYl6xgZM+9Js80WY1h9En8NtFPXmequkgyQJnsuUaVgxzZqoKmV73O/q9Q+rtyqxYB/NsMKTFThl9aX5I1VX9C7F8KUDc1x+CyDklEuLIn3tgBdrDb3eveNIFA==\\">
<pre><code class=\\"language-md\\"># Title

is very handsome.
</code></pre>
<pre><code class=\\"language-ts\\">const message: string = &quot;Mr.Hope&quot;;

document.querySelector(&quot;h1&quot;).innerHTML = message;
</code></pre>
<pre><code class=\\"language-scss\\">h1 {
  font-style: italic;

  + p {
    color: red;
  }
}
</code></pre>
</CodeDemo>"
`;
